OverlayPlugin.Core/EventSources/FFXIVRequiredEventSource.cs

```diff
@@ -0,0 +1,411 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+// For some reason this using is required by the github build?
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+using Advanced_Combat_Tracker;
+using FFXIV_ACT_Plugin.Common;
+using Newtonsoft.Json.Linq;
+using RainbowMage.OverlayPlugin.MemoryProcessors.Combatant;
+using RainbowMage.OverlayPlugin.NetworkProcessors;
+using PluginCombatant = FFXIV_ACT_Plugin.Common.Models.Combatant;
+
+namespace RainbowMage.OverlayPlugin.EventSources
+{
+    partial class FFXIVRequiredEventSource : EventSourceBase
+    {
+        private ReadOnlyCollection<uint> cachedPartyList = null;
+        private List<uint> missingPartyMembers = new List<uint>();
+        private bool ffxivPluginPresent = false;
+        private static Dictionary<uint, string> StatusMap = new Dictionary<uint, string>
+        {
+            { 0, "Online" },
+            { 12, "Busy" },
+            { 15, "InCutscene" },
+            { 17, "AFK" },
+            { 21, "LookingToMeld" },
+            { 22, "RP" },
+            { 23, "LookingForParty" },
+        };
+
+        private const string OnlineStatusChangedEvent = "OnlineStatusChanged";
+        private const string PartyChangedEvent = "PartyChanged";
+
+        private FFXIVRepository repository;
+        private ICombatantMemory combatantMemory;
+
+        // Event Source
+
+        public BuiltinEventConfig Config { get; set; }
+
+        public FFXIVRequiredEventSource(TinyIoCContainer container) : base(container)
+        {
+            Name = "FFXIVRequired";
+            var haveRepository = container.TryResolve(out repository);
+            var haveCombatantMemory = container.TryResolve(out combatantMemory);
+
+            if (haveRepository && haveCombatantMemory)
```

Comment:
My preference is that this sort of thing should just be an early out rather than a long indented conditional.

Reply:
This PR was a pure lift and shift of event handlers wherever possible, I'd rather not mix different refactoring in the same PR.